Drop-In/src/main/java/com/braintreepayments/api/DropInEvent.java

```diff
@@ -0,0 +1,116 @@
+package com.braintreepayments.api;
+
+import android.os.Bundle;
+import android.os.Parcel;
+import android.os.Parcelable;
+
+class DropInEvent implements Parcelable {
+
+    private final DropInEventType type;
+    private final Bundle payload;
+
+    static DropInEvent createSendAnalytics(String eventName) {
+        DropInEvent event = new DropInEvent(DropInEventType.SEND_ANALYTICS);
+        event.putString(DropInEventProperty.ANALYTICS_EVENT_NAME, eventName);
+        return event;
+    }
+
+    static DropInEvent createAddCardSubmitEvent(String cardNumber) {
+        DropInEvent event = new DropInEvent(DropInEventType.ADD_CARD_SUBMIT);
+        event.putString(DropInEventProperty.CARD_NUMBER, cardNumber);
+        return event;
+    }
+
+    static DropInEvent createCardDetailsSubmitEvent(Card card) {
+        DropInEvent event = new DropInEvent(DropInEventType.CARD_DETAILS_SUBMIT);
+        event.putParcelable(DropInEventProperty.CARD_DETAILS, card);
+        return event;
+    }
+
+    static DropInEvent createEditCardEvent(String cardNumber) {
+        DropInEvent event = new DropInEvent(DropInEventType.EDIT_CARD);
+        event.putString(DropInEventProperty.CARD_NUMBER, cardNumber);
+        return event;
+    }
+
+    static DropInEvent createDeleteVaultedPaymentMethodNonceEvent(PaymentMethodNonce paymentMethodNonceToDelete) {
+        DropInEvent event = new DropInEvent(DropInEventType.DELETE_VAULTED_PAYMENT_METHOD);
+        event.putParcelable(DropInEventProperty.VAULTED_PAYMENT_METHOD_SELECTION, paymentMethodNonceToDelete);
+        return event;
+    }
+
+    static DropInEvent createVaultedPaymentMethodSelectedEvent(PaymentMethodNonce paymentMethodNonce) {
+        DropInEvent event = new DropInEvent(DropInEventType.VAULTED_PAYMENT_METHOD_SELECTED);
+        event.putParcelable(DropInEventProperty.VAULTED_PAYMENT_METHOD_SELECTION, paymentMethodNonce);
+        return event;
+    }
+
+    static DropInEvent createSupportedPaymentMethodSelectedEvent(DropInPaymentMethodType type) {
+        DropInEvent event = new DropInEvent(DropInEventType.SUPPORTED_PAYMENT_METHOD_SELECTED);
+        event.putString(DropInEventProperty.SUPPORTED_PAYMENT_METHOD_SELECTION, type.name());
```

Comment:
Nitpick - but SELECTION doesn't feel like the right naming here. How do you feel about `VAULTED_PAYMENT_METHOD_NONCE` and `SUPPORTED_PAYMENT_METHOD_TYPE`

Reply:
+1 on the renaming. We can review and rename.